packages/rebber-plugins/src/preprocessors/codeVisitor.js

```diff
@@ -1,41 +1,53 @@
 const visit = require('unist-util-visit')
+const appendix = require('../type/appendix')
 
 module.exports = (ctx, tree) => {
-  const title = ctx.codeAppendiceTitle || 'Appendices'
+  ctx.overrides.appendix = appendix
   return (node) => {
-    const inAppendix = []
-    let appendixIndex = 1
-
+    let appendix = tree.children[tree.children.length - 1]
+    if (!appendix || appendix.type !== 'appendix') {
+      appendix = {
+        type: 'appendix',
+        children: [],
+      }
+      tree.children.push(appendix)
+    }
+    let appendixIndex = appendix.children.length + 1
     visit(node, 'code', (innerNode, index, _parent) => {
-      inAppendix.push({
+      appendix.children.push({
         type: 'paragraph',
         children: [
           {
-            type: 'definition',
-            identifier: `appendix-${appendixIndex}`,
-            referenceType: 'full',
-            children: [{type: 'text', value: 'code'}],
+            type: 'heading',
+            children: [
+              {
+                type: 'definition',
+                identifier: `appendix-${appendixIndex}`,
+                url: `${ctx.codeAppendiceTitle || 'Appendix'} ${appendixIndex}`,
+                referenceType: 'full',
+                children: [{type: 'text', value: 'code'}],
+              },
+            ],
+            depth: 1,
           },
-          {type: 'text', value: '\n'},
-          innerNode,
         ],
       })
+
+      appendix.children.push(innerNode)
+
       const referenceNode = {
         type: 'linkReference',
         identifier: `appendix-${appendixIndex}`,
+        children: [
+          {
+            type: 'text',
+            value: 'code',
```

Comment:
Please give another name than `code` because it is awfully unclear, maybe `Code : annexe ${appendixIndex}` is good.

Reply:
I was wondering : we could add another parameter "codeAppendinxLabel" to change this. This will avoid putting hard coded french word inside rebber-plugins module

Reply:
I'm fine with this option.

Reply:
Done for me